tests/template_tests/syntax_tests/test_basic.py

```diff
@@ -164,6 +164,88 @@ def test_basic_syntax19(self):
         else:
             self.assertEqual(output, "")
 
+    @setup({"basic-syntax19a": "{{ klass.in_template }}"})
+    def test_access_class_property(self):
+        class MyClass:
+            in_template = True
+            do_not_call_in_templates = True  # prevent instantiation
+
+            # Make class type subscriptable
+            def __class_getitem__(cls, key):
+                from types import GenericAlias
+
+                return GenericAlias(cls, key)
```

Comment:
```suggestion
```
I think we don't need this for this test (I might be wrong)
`GenericAlias` was adding in Python 3.9, so removing this gives us a test that passes on 3.8 but fails on 3.9 without the fix

Reply:
We need to either keep this or inherit from `list` or `dict` (I understood Carlton's comment that he was not happy that the `__class_getitem__` was not explicit in the tests). 

In Py3.8 `__class_getitem__` never gets called, `GenericAlias` never imported and the test should pass. In Py3.9+ it might get called (without the patch) and will fail.

Reply:
Can we have one test that inherits from `list`? 
These tests error for me on Python 3.8

```
ERROR: test_access_class_method (template_tests.syntax_tests.test_basic.BasicSyntaxTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/django/django/test/utils.py", line 461, in inner
    return func(*args, **kwargs)
  File "/django/tests/template_tests/utils.py", line 58, in inner
    func(self)
  File "/django/tests/template_tests/syntax_tests/test_basic.py", line 197, in test_access_class_method
    output = self.engine.render_to_string("basic-syntax19b", {"class_var": MyClass})
  File "/django/django/template/engine.py", line 195, in render_to_string
    return t.render(Context(context, autoescape=self.autoescape))
  File "/django/django/template/base.py", line 175, in render
    return self._render(context)
  File "/django/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
  File "/django/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/django/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/django/django/template/base.py", line 966, in render_annotated
    return self.render(context)
  File "/django/django/template/base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
  File "/django/django/template/base.py", line 715, in resolve
    obj = self.var.resolve(context)
  File "/django/django/template/base.py", line 847, in resolve
    value = self._resolve_lookup(context)
  File "/django/django/template/base.py", line 880, in _resolve_lookup
    current = current[bit]
  File "/django/tests/template_tests/syntax_tests/test_basic.py", line 189, in __class_getitem__
    from types import GenericAlias
ImportError: cannot import name 'GenericAlias' from 'types' (/home/sarahboyce/.pyenv/versions/3.8.19/lib/python3.8/types.py)

======================================================================
ERROR: test_access_class_property (template_tests.syntax_tests.test_basic.BasicSyntaxTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/django/django/test/utils.py", line 461, in inner
    return func(*args, **kwargs)
  File "/django/tests/template_tests/utils.py", line 58, in inner
    func(self)
  File "/django/tests/template_tests/syntax_tests/test_basic.py", line 179, in test_access_class_property
    output = self.engine.render_to_string("basic-syntax19a", {"class_var": MyClass})
  File "/django/django/template/engine.py", line 195, in render_to_string
    return t.render(Context(context, autoescape=self.autoescape))
  File "/django/django/template/base.py", line 175, in render
    return self._render(context)
  File "/django/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
  File "/django/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/django/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/django/django/template/base.py", line 966, in render_annotated
    return self.render(context)
  File "/django/django/template/base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
  File "/django/django/template/base.py", line 715, in resolve
    obj = self.var.resolve(context)
  File "/django/django/template/base.py", line 847, in resolve
    value = self._resolve_lookup(context)
  File "/django/django/template/base.py", line 880, in _resolve_lookup
    current = current[bit]
  File "/django/tests/template_tests/syntax_tests/test_basic.py", line 175, in __class_getitem__
    from types import GenericAlias
ImportError: cannot import name 'GenericAlias' from 'types' (/home/sarahboyce/.pyenv/versions/3.8.19/lib/python3.8/types.py)

======================================================================
ERROR: test_access_class_property_returned_by_callable (template_tests.syntax_tests.test_basic.BasicSyntaxTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/django/django/test/utils.py", line 461, in inner
    return func(*args, **kwargs)
  File "/django/tests/template_tests/utils.py", line 58, in inner
    func(self)
  File "/django/tests/template_tests/syntax_tests/test_basic.py", line 215, in test_access_class_property_returned_by_callable
    output = self.engine.render_to_string(
  File "/django/django/template/engine.py", line 195, in render_to_string
    return t.render(Context(context, autoescape=self.autoescape))
  File "/django/django/template/base.py", line 175, in render
    return self._render(context)
  File "/django/django/test/utils.py", line 112, in instrumented_test_render
    return self.nodelist.render(context)
  File "/django/django/template/base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/django/django/template/base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/django/django/template/base.py", line 966, in render_annotated
    return self.render(context)
  File "/django/django/template/base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
  File "/django/django/template/base.py", line 715, in resolve
    obj = self.var.resolve(context)
  File "/django/django/template/base.py", line 847, in resolve
    value = self._resolve_lookup(context)
  File "/django/django/template/base.py", line 880, in _resolve_lookup
    current = current[bit]
  File "django/tests/template_tests/syntax_tests/test_basic.py", line 207, in __class_getitem__
    from types import GenericAlias
ImportError: cannot import name 'GenericAlias' from 'types' (/home/sarahboyce/.pyenv/versions/3.8.19/lib/python3.8/types.py)
```

Reply:
Dear, oh dear. It seems `__class_getitem__` was introduced before `GenericAlias`. I just reverted the test to subclass `list`. Makes it shorter and I mention `__class_getitem__` in the comments. 🤔